pkg/api/payer/clientManager.go

```diff
@@ -0,0 +1,59 @@
+package payer
+
+import (
+	"sync"
+
+	"github.com/xmtp/xmtpd/pkg/registry"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+/*
+*
+The ClientManager contains a mapping of nodeIDs to gRPC client connections.
+
+These client connections are safe to be shared and re-used and will automatically attempt
+to reconnect if the underlying socket connection is lost.
+*
+*/
+type ClientManager struct {
+	log          *zap.Logger
+	connections  sync.Map // map[uint32]*grpc.ClientConn
+	nodeRegistry registry.NodeRegistry
+}
+
+func NewClientManager(log *zap.Logger, nodeRegistry registry.NodeRegistry) *ClientManager {
+	return &ClientManager{log: log, nodeRegistry: nodeRegistry}
+}
+
+func (c *ClientManager) GetClient(nodeID uint32) (*grpc.ClientConn, error) {
+	existing, ok := c.connections.Load(nodeID)
+	if ok {
+		return existing.(*grpc.ClientConn), nil
+	}
+
+	conn, err := c.newClientConnection(nodeID)
+	if err != nil {
+		return nil, err
+	}
+	// Store the connection
+	c.connections.Store(nodeID, conn)
```

Comment:
I noticed that we're using a `sync.Map` here, implying some degree of concurrency. Just noting that we seem to have a `write-after-read` race condition here, which may or may not matter depending on what you're trying to use this for

Reply:
It shouldn't really matter, but it's easy enough to fix so that it's bulletproof. Maybe will prevent some future error, so I updated it.